packages/mobile/src/redux/reducers.ts

```diff
@@ -40,7 +42,23 @@ export default combineReducers({
   recipients,
   localCurrency,
   imports,
-}) as () => RootState
+}) as (state: RootState | undefined, action: Action) => RootState
+
+const rootReducer = (state: RootState | undefined, action: Action): RootState => {
+  if (action.type === Actions.CLEAR_STORED_ACCOUNT && state) {
+    // Generate an initial state but keep the information not specific to the account
+    // that we want to save.
+    const initialState = appReducer(undefined, action)
+    return {
+      ...initialState,
+      localCurrency: state.localCurrency,
+      identity: identity(state.identity, deleteUserData()),
```

Comment:
Is there anything else important to keep?

Reply:
Like I said above, I feel we shouldn't keep anything inside `identity`.

Reply:
It's worth discussing, let's chat about it on slack?